providers/base/tests/test_edid_cycle.py

```diff
@@ -0,0 +1,164 @@
+"""This module provides test cases for the edid_cycle module."""
+import unittest
+from unittest.mock import patch, call, Mock
+
+from bin import edid_cycle
+
+
+class ZapperEdidCycleTests(unittest.TestCase):
+    """This class provides test cases for the edid_cycle module."""
+
+    @patch("time.sleep", new=Mock)
+    @patch("bin.edid_cycle.zapper_run", new=Mock)
+    @patch("builtins.open")
+    @patch("os.getenv")
+    @patch("subprocess.check_output")
+    def test_discover_video_output_device(
+        self, mock_check, mock_getenv, mock_open
+    ):
+        """
+        Check if the function automatically discover the port
+        under test with output given by randr in different conditions.
+        """
+        disconnected_output = {"x11": "Monitors: 0", "wayland": ""}
+
+        connected_output = {
+            "x11": (
+                "Monitors: 1\n"
+                " 0: +HDMI-1 1920/576x1080/324+800+1080 HDMI-1\n"
+            ),
+            "wayland": (
+                "x: 0, y: 0, scale: 1, rotation: normal, primary: yes\n"
+                "associated physical monitors:\n"
+                "    HDMI-1 TSB PI-KVM Video 0x88888800\n\n"
+                "HDMI-1 TSB PI-KVM Video\n"
```

Comment:
Minor: `textwrap.dedent("""mutlilinestring instead of concatenating via \n""")` looks a bit better

Reply:
TIL, thanks